### DISCUSSION TITLE
brew upgrade libx11 fails from source

### DISCUSSION THREAD

**Discussion by Yann-R** [Homebrew/discussions#4164] [Everyday usage] [answered]
### Output of `brew config`

```shell
$ brew config
HOMEBREW_VERSION: 3.6.20-169-g856a347
ORIGIN: https://github.com/Homebrew/brew
HEAD: 856a347bfd0aa05ad644a864edf311ea1f7cbbf8
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9d2fca0f48c1a7966a141aad47456dfb92bc934c
Core tap last commit: 8 days ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/brew
HOMEBREW_REPOSITORY: /opt/brew/Homebrew
HOMEBREW_CELLAR: /opt/brew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.1QywUNaafY/org.xquartz:0
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.39.1 => /opt/brew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.2-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  xtail
  growly
  colorsvn
  contacts
  godep
  csv-fix
  szip

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  brew-pip
  gcc@6
  godep
  isl@0.18
  q
  rdesktop

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/ublio.h

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  unison

Warning: Some installed formulae are not readable:
  growly: wrong number of arguments (given 1, expected 0)

Warning: Your Homebrew's prefix is not /usr/local.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from MacHomebrew on Twitter.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

$ brew upgrade libx11
==> Upgrading 1 outdated package:
libx11 1.8.2 -> 1.8.3
==> Fetching libx11
Warning: Building libx11 from source as the bottle needs:
  HOMEBREW_CELLAR: /usr/local/Cellar (yours is /opt/brew/Cellar)
  HOMEBREW_PREFIX: /usr/local (yours is /opt/brew)
==> Downloading https://www.x.org/archive/individual/lib/libX11-1.8.3.tar.gz
Already downloaded: /Users/ricquebo/Library/Caches/Homebrew/downloads/ca14e5467e84cea121a2d08edb9bc6e7acfe26a519ddf1e25957c0ebbe062c7f--libX11-1.8.3.tar.gz
==> Upgrading libx11
  1.8.2 -> 1.8.3 

==> ./configure --prefix=/opt/brew/Cellar/libx11/1.8.3 --sysconfdir=/opt/brew/etc --localstatedir=/opt/brew/var --enable-unix-transport --enable-tcp-transport --enable-ipv6 --enable-local-transport --enable-loadable-i18n --enable-xthreads --enable-specs=no
Last 15 lines from /Users/ricquebo/Library/Logs/Homebrew/libx11/01.configure:
checking for BIGFONT... yes
checking for getpagesize... yes
checking for working mmap... yes
checking for nl_langinfo... yes
checking for X11... no
configure: error: Package requirements (xproto >= 7.0.25 xextproto xtrans xcb >= 1.11.1 kbproto inputproto) were not met:

Package 'pthread-stubs', required by 'xcb', not found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables X11_CFLAGS
and X11_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

Do not report this issue to Homebrew/brew or Homebrew/core!

**Comment by SMillerDev** [comment]
> Package 'pthread-stubs', required by 'xcb', not found

Do you have this installed?

**Comment by gromgit** [comment]
>```
>Package 'pthread-stubs', required by 'xcb', not found
>```

This was fixed in https://github.com/Homebrew/homebrew-core/pull/111928. You'll have to `brew reinstall xcb` to grab that particular fix, though.

**Comment by Yann-R** [comment, marked as answer]
Oh, great.
Indeed my previous `libx11-1.8.2` did not install `xcb` and its upgrade to 1.8.3 did not imply this either.
But a `brew reinstall libxcb` got `xcb`... and then installing `libx11` is successful.
Your rock!
Thank

**Reply by kojix2** [reply]
`brew reinstall libxcb` worked fine. Thanks.
